### ISSUE TITLE
read.ctd() misinterpreting originalNames of repeated-component data

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1141]
For example,
```R
library(oce)
f <- system.file("extdata", "d201211_0011.cnv", package="oce")
d <- read.oce(f)
summary(d)
```
gives (other stuff trimmed)
```
                                      Min.       Mean    Max.   Dim. OriginalName      
temperature [°C, ITS-90]              -0.0155    2.202   4.5222 78   t190C             
temperature2 [°C, ITS-90]             -0.0104    2.2086  4.536  78   -                 
```
and I know from inspection of the file that `temperature` was originally called `t090c` and `temperature2` as originally called `t190C`.

**Comment by dankelley** [comment]
Once I fix this, I will put in some a build-test to ensure that all data get proper originalNames; that way, oce will fail to build if this problem comes back. I will also check the numerical values of the first couple of data points, just to be extra-safe.

**Comment by dankelley** [comment]
Fixed in `develop` commit c74ab47b6e54e416ffca17d3119cd7635df17b09 and now there is a build-test to ensure that this does not come up again.  I should have coded this solution a long time ago, instead of fiddling with two schemes that were faulty. Sometimes a loop does the trick!
